### PR TITLE
Added to pass ssl_mode in configuration.

### PR DESCRIPTION
**Pull request by migimigi**
SSL was enabled by default when connecting to MySQL 5.7.
See https://dev.mysql.com/worklog/task/?id=7712

Added to configuration to enable/disable SSL.
The reason for not using --ssl or --skip-ssl described in MySQL Worklog, --ssl option is deprecated as of MySQL 5.7.11.
See https://dev.mysql.com/doc/refman/5.7/en/encrypted-connection-options.html

### CODE REVIEW

**Comment by methane** [comment]
I think there is no MYSQL_OPT_SSL_MODE in old MySQL Connector/C or MariaDB.
Would you test it?

**Comment by migimigi** [comment]
MYSQL_OPT_SSL_MODE exists from MySQL5.5.55.
https://dev.mysql.com/doc/relnotes/mysql/5.5/en/news-5-5-55.html#mysqld-5-5-55-security
I don't know about MariaDB.
I'm going to investigate, fix and test.
I couldn't find MySQL and MariaDB versions supported by mysqlclient-python at https://mysqlclient.readthedocs.io/
Is it there?

**Comment by methane** [comment]
MySQL 5.5+ are supported.

**Comment by methane** [comment]
I used MYSQL_DEFAULT_AUTH which is introduced in MySQL 5.5.10.  (see #396)
So MySQL 5.5.10+ are supported now.

**Comment by migimigi** [comment]
I see.

**Comment by migimigi** [comment]
The version in which MYSQL_OPT_SSL_MODE is defined
5.5.55- https://dev.mysql.com/doc/relnotes/mysql/5.5/en/news-5-5-55.html#mysqld-5-5-55-security
5.6.36- https://dev.mysql.com/doc/relnotes/mysql/5.6/en/news-5-6-36.html#mysqld-5-6-36-security
5.7.11- https://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-11.html#mysqld-5-7-11-security
5.8.0- https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-0.html#mysqld-8-0-0-security
Other than these versions were disabled with #if.

MYSQL_OPT_SSL_MODE was found to be not defined in MariaDB.
So it was disabled in MariaDB.
https://github.com/MariaDB/server/blob/10.5/include/mysql.h#L184
https://github.com/brianmario/mysql2/issues/879#issuecomment-437951093

**Comment by methane** [comment]
Options should be similar to `my.cnf` file.  Imagine that users may load options from their config file.
So `ssl_mode="VERIFY_CA"`, not `ssl_mode=SSL_MODE.VERIFY_CA`.

**Comment by migimigi** [comment]
It has modified

**Comment by methane** [comment]
I don't like this.  This implementation ignores the ssl_mode option silently when it is not supported by the underlying library.

There are two options to fix this:

* Raise an error if ssl_mode is not supported by the underlying library.
* Support ssl_mode using legacy options.

**Comment by migimigi** [comment]
I see.
I select to raises an error if the ssl_mode setting is not supported,
didn't choose the legacy option because it was deprecated of 5.7.11 or later

File: MySQLdb/_mysql.c [hunk 1]
```diff
@@ -2567,6 +2592,12 @@ _mysql_methods[] = {
         METH_NOARGS,
         _mysql_get_client_info__doc__
     },
+    {
+        "get_have_enum_mysql_opt_ssl_mode",
+        (PyCFunction)_mysql_get_have_enum_mysql_opt_ssl_mode,
+        METH_NOARGS,
+        _mysql_get_have_enum_mysql_opt_ssl_mode__doc__
+    },
```
Comment (methane) [inline MySQLdb/_mysql.c]: Don't add new API.

File: MySQLdb/connections.py [hunk 2]
```diff
@@ -102,6 +102,13 @@ class object, used to create cursors (keyword only)
         :param int client_flag:
             flags to use or 0 (see MySQL docs or constants/CLIENTS.py)
 
+        :param str ssl_mode
```
Comment (methane) [inline MySQLdb/connections.py]:
```suggestion
        :param str ssl_mode:
```

File: MySQLdb/connections.py [hunk 3]
```diff
@@ -162,6 +169,14 @@ class object, used to create cursors (keyword only)
...
         kwargs2['client_flag'] = client_flag
 
+        if 'ssl_mode' in kwargs:
+            if not _mysql.get_have_enum_mysql_opt_ssl_mode():
+                raise NotSupportedError('Does not support ssl_mode specification: %s' % _mysql.get_client_info())
+            if hasattr(SSL_MODE, kwargs['ssl_mode']):
+                kwargs2['ssl_mode'] = getattr(SSL_MODE, kwargs['ssl_mode'])
+            else:
+                raise NotSupportedError('Unknown MySQL ssl_mode specification: %s' % kwargs['ssl_mode'])
+
```
Comment (methane) [inline MySQLdb/connections.py]: Implement this block in C.

File: MySQLdb/_mysql.c (hunk 1 above)
Comment (migimigi) [reply MySQLdb/_mysql.c]: I see.

File: MySQLdb/connections.py (hunk 2 above)
Comment (migimigi) [reply MySQLdb/connections.py]: thanks

File: MySQLdb/connections.py (hunk 3 above)
Comment (migimigi) [reply MySQLdb/connections.py]: Implemented error checking on the C lang side.
And removed MySQLdb.constants.SSL_MODE, which is no longer needed.

**Comment by methane** [comment]
Thanks!

**Pull request merged**